components/ocmcli/Makefile

```
@@ -68,40 +75,35 @@ $(GEN)/build: $(GEN) $(GEN)/.exists $(CMDSRCS) $(OCMSRCS)
 	@touch $(GEN)/build
 
 .PHONY: image
-image: $(GEN)/image
-
-$(GEN)/image: $(GEN)/.exists Dockerfile $(GEN)/build
+image:
 	@PLATFORM_OS_OVERRIDE=$(PLATFORM_OS); \
 	if [ $$PLATFORM_OS_OVERRIDE == darwin ]; then \
 		echo; echo "Building linux instead of darwin as there's no native Docker platform for darwin"; echo; \
-		docker buildx build -t $(IMAGE):$(VERSION) --platform linux/$(PLATFORM_ARCH) --file Dockerfile $(REPO_ROOT) \
-			--build-arg OCM_VERSION=$(EFFECTIVE_VERSION) \
-			--build-arg SUFFIX=$$(echo linux/$(PLATFORM_ARCH) | sed -e s:/:-:g); \
+		docker buildx build -t $(IMAGE):$(VERSION) --platform linux/$(PLATFORM_ARCH) --file $(REPO_ROOT)/Dockerfile $(REPO_ROOT) \
+			--label org.opencontainers.image.version=$(VERSION) \
+			--label org.opencontainers.image.revision=$(VERSION) $(ADDITIONAL_BUILDX_ARGS); \
 	else \
 		echo; echo "Building for $(PLATFORM_OS)/$(ARCH)"; echo; \
-		docker buildx build -t $(IMAGE):$(VERSION) --platform $(PLATFORM_OS)/$(PLATFORM_ARCH) --file Dockerfile $(REPO_ROOT) \
-			--build-arg OCM_VERSION=$(EFFECTIVE_VERSION) \
-			--build-arg SUFFIX=$$(echo $(PLATFORM_OS)/$(PLATFORM_ARCH) | sed -e s:/:-:g); \
+		docker buildx build -t $(IMAGE):$(VERSION) --platform $(PLATFORM_OS)/$(PLATFORM_ARCH) --file $(REPO_ROOT)/Dockerfile $(REPO_ROOT) \
+			--label org.opencontainers.image.version=$(VERSION) \
+			--label org.opencontainers.image.revision=$(VERSION) $(ADDITIONAL_BUILDX_ARGS); \
 	fi
-	@touch $(GEN)/image
 
 .PHONY: image.multi
-image.multi: $(GEN)/image.multi
-
-$(GEN)/image.multi: Dockerfile $(GEN)/build
+image.multi:
 	for i in $(IMAGE_PLATFORMS); do \
-	tag=$$(echo $$i | sed -e s:/:-:g); \
-        echo building platform $$i; \
-	docker buildx build --load -t $(IMAGE):$(VERSION)-$$tag --platform $$i --file Dockerfile $(REPO_ROOT) \
-          --build-arg OCM_VERSION=$(EFFECTIVE_VERSION) \
-          --build-arg SUFFIX=$$tag; \
+		tag=$$(echo $$i | sed -e s:/:-:g); \
+			echo building platform $$i; \
+		docker buildx build --load -t $(IMAGE):$(VERSION)-$$tag --platform $$i --file $(REPO_ROOT)/Dockerfile $(REPO_ROOT) \
+			--build-arg OCM_VERSION=$(EFFECTIVE_VERSION) \
+			--label org.opencontainers.image.version=$(VERSION) \
+			--label org.opencontainers.image.revision=$(VERSION) $(ADDITIONAL_BUILDX_ARGS); \
```

```suggestion
			--label org.opencontainers.image.revision=$(VERSION) $(ADDITIONAL_BUILDX_ARGS)
```

These should not be required, right?

---

you need this to parallelize the build instruction

---

interesting. Thanks for clarification